projects/reliable-integration/0002-low-level-design.md

```diff
@@ -6,13 +6,49 @@ This low-level design intends to go deeper on the technical aspects of the proje
+| `Fresh start`                     | None                     | None                   | Empty          | Generate all events, from `event_id = 0`. All events come during the `SYNC` phase.                                                        |
+| `Snapshot start`                  | Some                     | None                   | Empty          | Generate all events, from `event_id = 0`. Events come during both the `LOAD` and `SYNC` phases.                                           |
+| `Restart from aborted LOAD phase` | Some                     | Some                   | `LOAD`         | All events in the database are discarded. Generate all events, from `event_id = 0`. Events come during both the `LOAD` and `SYNC` phases. |
+| `Restart from aborted SYNC phase` | Some                     | Some                   | `SYNC`         | Generate events from `event_id = last_event_id + 1`. All events come during the `SYNC` phase.                                             |
+
+There's no case in which there are some events in the database, but no vertices.
+
+For the `Fresh start` scenario, if we consider there's a `LOAD` phase with 0 events, then the desired outcome becomes the same for the first three scenarios. This makes it possible to simplify the table into only two scenarios, only depending on `State metadata`, as described in the beginning of this section:
+
+| Starting scenario                                                      | State metadata  | Desired outcome                                                                                                                           |
+|------------------------------------------------------------------------|-----------------|-------------------------------------------------------------------------------------------------------------------------------------------|
+| `Fresh start` or `Snapshot start` or `Restart from aborted LOAD phase` | Empty or `LOAD` | All events in the database are discarded. Generate all events, from `event_id = 0`. Events come during both the `LOAD` and `SYNC` phases. |
+| `Restart from aborted SYNC phase`                                      | `SYNC`          | Generate events from `event_id = last_event_id + 1`. All events come during the `SYNC` phase.                                             |
+
+Note: when generating snapshots, events should not be included, as they will be discarded and regenerated anyway.
+
+Now, we can describe a step-by-step representing these rules:
+
+1. The full node begins the initialization process
+2. The `EventManager` starts
+   1. It retrieves the previous node state from the database and stores it
+   2. If its stored previous state is empty or `LOAD`, it discards all events currently in the database
```

Comment:
Maybe, after clearing the database, it should add an event of "RESET EVENTS", so an integration can react accordingly to this case. Another point might be to keep the event id instead of resetting it to zero. What do you think?

Reply:
If we use a UUID for the "event database" and we emit a "event system start" with that UUID, we can have a system that does not have to know internally if it was reset or not. I think that would be better than making the "clearing of a database" something that needs to carefully emit a "reset event". What do you think?

Reply:
I agree with @jansegre , and also that UUID could be required to retrieve events. If the client requests events (via HTTP or WebSocket) using the wrong UUID, the full node rejects it, so the the client knows it has to process events from the beginning.

Reply:
Good approach! So, if I got it right, we will emit a "system start" event, right?

Reply:
Something like it, but it's more an "event system start" than a "system start". It would only be emitted when the event feature is started.

Reply:
I'll add this mechanism in a new code PR